devicehive-frontend/src/main/java/com/devicehive/security/util/JwtTokenGenerator.java

```diff
@@ -54,7 +54,8 @@
     public String generateToken(JwtPayload payload, TokenType tokenType) {
         
         long maxAge = tokenType.equals(TokenType.ACCESS) ? accessTokenMaxAge : refreshTokenMaxAge;
-        Date expiration = new Date(System.currentTimeMillis() + maxAge);
+        Date expiration = payload.getExpiration() != null ? payload.getExpiration() :
+                new Date(System.currentTimeMillis() + maxAge);
```

Comment:
Would it be better to use https://github.com/devicehive/devicehive-java-server/blob/7b9169b1cc6b0f619ef38245c7b625c003b6b003/devicehive-frontend/src/main/java/com/devicehive/service/time/LocalTimestampService.java for Date generation?